### PR TITLE
feat(#18): Booking Restrictions Per Room

### LINKED ISSUE #18: Booking Period Restrictions
### User story
As a user, I cannot make a reservation if the current date does not fall within the booking period for that room. 

### Narrative
As a user, it shouldn't be possible to reserve things too early, as this would allow people to just over book everything months in advance. For example, the Mezzanine, cannot be booked more than fourteen days in advance. 

As an administrator, I must be able to configure a room's booking period. For example, Room A must be booked 10 business days in advance, but cannot be booked more than 2 months in advance. 

### Out of scope
Restrictions unrelated to the booking window

### Acceptance criteria

| # | Given | When | Then |
|---|-------|------|------|
|1|I am able to book rooms|I attempt to make a reservation and the current date is within the booking window|The system makes the reservation|
|2|I am able to book rooms|I attempt to make a reservation and the current date is not within the booking window|The system doe not make the reservation|

### UI Mockup:
Booking Restriction

### PR DESCRIPTION
**Pull request by Alex-Hayeur**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**
Please provide enough information so that others can review your pull request:

Allows admins to define how early and how late in advance rooms can be booked. 
This means that rooms can't be booked earlier that "X" days in advance, or later than "Y" days in advance. 

C1
C2

Also, updated the ReadMe with information on testing. 

### Test plan (required)
Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

When rooms have a minimum and maximum days in advance to book, if a booking is made outside of that window it fails and an error message is given to the user explaining the exact booking window of that room. 

### Closing issues
List all issues this PR resolves.
- closes #18
### Pre-Merge checklist 
Complete these items before requesting reviews
- [x] Static Analysis Rules Validated
- [x] All commits follow naming conventions 

### Sub-tasks Checklist
- [X] UI Implementation
- [X] UI Test
- [x] Backend Implementation
- [x] Backend Tests
- [x] Integration Tests
- [ ] System Tests
- [x] Documentation

### CODE REVIEW

**Review by WindOfRussia — APPROVED** [review]
I've reviewed alex's code, updated the tests to pass and refactored a bit of the existing code as well.

**Review by alexstojda — DISMISSED** [review]
Code looks good! 

Please update the relevant acceptance tests with videos of the acceptance tests

File: resources/js/Pages/Admin/Rooms/UpdateRoomForm.vue [hunk 1]
```diff
@@ -159,6 +186,8 @@ export default {
             this.form.floor = room?.floor;
             this.form.building = room?.building;
             this.form.status = room?.status;
+            this.form.status = room?.min_days_advance;
+            this.form.status = room?.max_days_advance;
```
Comment (jonathanp16) [inline resources/js/Pages/Admin/Rooms/UpdateRoomForm.vue:190]: Should match the min and max variables in the form

**Review by jonathanp16 — CHANGES_REQUESTED** [review]
Looks great, just that one small issue in the update room form that I mentioned in the comments

**Review by jonathanp16 — APPROVED** [review]

File: resources/js/Pages/Admin/Rooms/UpdateRoomForm.vue (hunk 1 above)
Comment (Alex-Hayeur) [reply resources/js/Pages/Admin/Rooms/UpdateRoomForm.vue]: Addressed, thanks for catching that.

**Review by WindOfRussia — APPROVED** [review]
yas

**Pull request merged**